lib/sycamore/sycamore/transforms/merge_elements.py

```diff
@@ -135,7 +135,151 @@ def merge(self, elt1: Element, elt2: Element) -> Element:
+            return False
+
+        # DO NOT MERGE across pages
+        if not self.merge_across_pages and element1.properties["page_number"] != element2.properties["page_number"]:
+            return False
+
+        if element1.data["token_count"] + 1 + element2.data["token_count"] > self.max_tokens:
+            return False
+        
+        # MERGE adjacent 'text' elements (but not across pages - see above)
+        if ((element1 != None) and (element1.type == 'Text') and 
+            (element2 != None) and (element2.type == 'Text')):
+            return True
+        
+        # MERGE 'Section-header' + 'table' 
+        if ((element1 != None) and (element1.type == 'Section-header') and 
+            (element2 != None) and (element2.type == 'table')):
+            return True
+        
+        # Merge 'image' + 'text'* until next 'image' or 'Section-header' 
+        if ((element1 != None) and (element1.type == 'Image' or element1.type == 'Image+Text') and 
+            (element2 != None) and (element2.type == 'Text')):
+            return True
+        return False
+                
+    def merge(self, elt1: Element, elt2: Element) -> Element:
+        """Merge two elements; the new element's fields will be set as:
+            - type: "Section"
+            - binary_representation: elt1.binary_representation + elt2.binary_representation
+            - text_representation: elt1.text_representation + elt2.text_representation
+            - bbox: the minimal bbox that contains both elt1's and elt2's bboxes
+            - properties: elt1's properties + any of elt2's properties that are not in elt1
+            note: if elt1 and elt2 have different values for the same property, we take elt1's value
+            note: if any input field is None we take the other element's field without merge logic
+
+        Args:
+            element1 (Tuple[Element, int]): the first element (and number of tokens in it)
+            element2 (Tuple[Element, int]): the second element (and number of tokens in it)
+
+        Returns:
+            Tuple[Element, int]: a new merged element from the inputs (and number of tokens in it)
+        """
+
+        tok1 = elt1.data["token_count"]
+        tok2 = elt2.data["token_count"]
+        new_elt = Element()
+        # 'text' + 'text' = 'text'
+        # 'image' + 'text' = 'image+text'
+        # 'image+text' + 'text' = 'image+text'
+        # 'Section-header' + 'table' = 'Section-header+table'
+
+        if (elt1.type == "Image" or elt1.type == "Image+Text") and elt2.type == "Text":
+            new_elt.type = "Image+Text"
+        elif elt1.type == "Text" and elt2.type == "Text":
+            new_elt.type = "Text"
+        elif elt1.type == "Section-header" and elt2.type == "table":
+            new_elt.type = "Section-header+table"
+        else:
+            new_elt.type = "????"
+         
+        # Merge binary representations by concatenation
```

Comment:
Is this ever useful? I confess I don't have a better way to merge binary in general, but I can't imagine this resulting in something useable.

Reply:
Yes, that makes sense. I guess the only case where this might be marginally useful is where the binary_representation is actual text (when reading from a file), but even then the text_representation should be populated. I will remove the concatenation. Will replace with keeping the binary of the Image element.